packages/material-ui/src/useMediaQuery/useMediaQuery.test.js

```diff
@@ -51,142 +52,154 @@ describe('useMediaQuery', () => {
 
   describe('option: defaultMatches', () => {
     it('should be false by default', () => {
+      const ref = React.createRef();
+      const text = () => ref.current.textContent;
       const Test = () => {
         const matches = useMediaQuery('(min-width:2000px)');
-        values(matches);
-        return <span>{`${matches}`}</span>;
+        React.useEffect(() => values(matches), [matches]);
```

Comment:
global side-effect -> React.useEffect

Reply:
We are asserting the number of renders. What side effect are we talking about?

Reply:
It's important here as we try to minimize the number of renders. It's important when useMediaQuery is used frequently in a codebase.

Reply:
> What side effect are we talking about?

calinng `values` which is a side-effect.

Reply:
`values` is a spy, can we be pragmatic as opposed to dogmatic? As long as we keep a test that make sure we minimize the number of render, I'm happy. It can be a new one.

Reply:
I don't understand. The assertion is still the same. What kind of regression would this catch?

Reply:
We won't catch the case were it renders twice with the matches value.

```suggestion
        React.useEffect(() => values(matches));
```